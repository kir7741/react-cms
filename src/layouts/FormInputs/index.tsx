import React from 'react';
import classnames from 'classnames';
import Input from 'components/atoms/Input';
import Textarea from 'components/atoms/Textarea';
import PasswordInput from 'components/atoms/PasswordInput';
import DropdownSelect from 'components/atoms/DropdownSelect';
import useForm from 'util/hook/useForm';
import { FormControl, FormControlBase } from 'types/interfaces/form-control';
import { fakeOptions } from 'types/constants/fake-options';
import { RadioGroup } from 'components/atoms/RadioGroup';
import { CheckboxGroup } from 'components/atoms/CheckboxGroup';

import globalStyles from 'global.css';
import styles from './index.css';

interface FormInputs {
	textInput: FormControl<string>;
	numberInput: FormControl<number>;
	disableInput: FormControl<string>;
	passWordInput: FormControl<string>;
	subInput1: FormControl<string>;
	subInput2: FormControl<string>;
	textarea: FormControl<string>;
	selectInput: FormControl<string>;
	radioInput: FormControl<string>;
	checkboxInput: FormControl<boolean[]>;
}

const FormInputs: React.FC = () => {

	const [
		{ form },
		{ setCtrlValue, updateCtrlValidity, updateValidity, getCtrlErrorMsg}
	] = useForm<FormControlBase<FormInputs>>({
		textInput: {
			value: '',
			errors: null,
			options: {
				validators: []
			}
		},
		numberInput: {
			value: 0,
			errors: null,
			options: {
				validators: []
			}
		},
		disableInput: {
			value: '',
			errors: null,
			options: {
				validators: []
			}
		},
		passWordInput: {
			value: '',
			errors: null,
			options: {
				validators: []
			}
		},
		subInput1: {
			value: '',
			errors: null,
			options: {
				validators: []
			}
		},
		subInput2: {
			value: '',
			errors: null,
			options: {
				validators: []
			}
		},
		textarea: {
			value: '',
			errors: null,
			options: {
				validators: []
			}
		},
		selectInput: {
			value: '',
			errors: null,
			options: {
				validators: []
			}
		},
		radioInput: {
			value: 'M',
			errors: null,
			options: {
				validators: []
			}
		},
		checkboxInput: {
			value: [false , false, true],
			errors: null,
			options: {
				validators: []
			}
		}

	})

	return (
		<div className={classnames(styles.formInputs, globalStyles.row2)}>
			<div className={styles.leftPart}>
				<Input
					type="text"
					placeholder='請輸入'
					value={form.textInput.value}
					errorMsg=''
					onChangeValue={val => setCtrlValue('textInput', val)}
				/>
				<Input
					type="number"
					placeholder='請輸入'
					value={form.numberInput.value}
					errorMsg=''
					onChangeValue={val => setCtrlValue('numberInput', val)}
				/>
				<Input
					disabled
					type="text"
					placeholder='請輸入'
					value={form.disableInput.value}
					errorMsg=''
					onChangeValue={val => setCtrlValue('disableInput', val)}
				/>
				<PasswordInput
					placeholder='密碼'
					value={form.passWordInput.value}
					validOnBlur
					onChangeValue={val => setCtrlValue('passWordInput', val)}
				/>

				<div className={classnames(globalStyles.row2)}>
					<Input
						type="text"
						placeholder='請輸入'
						value={form.subInput1.value}
						errorMsg=''
						onChangeValue={val => setCtrlValue('subInput1', val)}
					/>
					<Input
						type="text"
						placeholder='請輸入'
						value={form.subInput2.value}
						errorMsg=''
						onChangeValue={val => setCtrlValue('subInput2', val)}
					/>
				</div>

				<Textarea
					value={form.textarea.value}
					placeholder='請輸入'
					className={styles.textarea}
					onBlur={() => {}}
					onChangeValue={val => setCtrlValue('textarea', val)}
				/>

			</div>

			<div className={styles.rightPart}>
				<DropdownSelect
					options={fakeOptions}
					selectedId={form.selectInput.value}
					onChangeValue={val => setCtrlValue('selectInput', val)}
				/>

				<RadioGroup
					value={form.radioInput.value}
					styleMap={({
						radioGroup: styles.radioGroup,
						radio: styles.radio
					})}
					name='radioInput'
					errorMsg=''
					options={[{id: 'M', name: '男'}, {id: 'F', name: '女'}]}
					onChangeValue={val => setCtrlValue('radioInput', val)}
				/>

				<CheckboxGroup
					value={form.checkboxInput.value}
					styleMap={({
						checkboxGroup: styles.checkboxGroup,
						checkbox: styles.checkbox
					})}
					errorMsg=''
					options={['選項A', '選項B', '選項C']}
					onGroupChangeValue={val => setCtrlValue('checkboxInput', val)}
				/>
			</div>

		</div>
	)
};

export default FormInputs;
