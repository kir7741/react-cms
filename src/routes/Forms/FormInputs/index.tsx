import React from 'react';
import { CustomRoute } from 'util/hook/useRouter';

const FormInputsRoutes: CustomRoute = {
	path: '/formsInputs',
	components: () => [import(/* webpackChunkName: 'members' */ './component')],
	render: ([FormInputs]) => <FormInputs />,
	onEnter: async ({ next }) => {
		const children = await next();
		return children;
	}
};
export default FormInputsRoutes;
