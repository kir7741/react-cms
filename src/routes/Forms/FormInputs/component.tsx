import FormInputs from 'layouts/FormInputs';
import { hot } from 'react-hot-loader/root';

export default hot(FormInputs);
