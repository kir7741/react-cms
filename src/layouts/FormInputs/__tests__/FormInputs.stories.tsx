import React from 'react';
import { storiesOf } from '@storybook/react';

import FormInputs from 'layouts/FormInputs';

const stories = storiesOf('layouts/FormInputs', module);

stories.add('__interactive', () => <FormInputs />);
