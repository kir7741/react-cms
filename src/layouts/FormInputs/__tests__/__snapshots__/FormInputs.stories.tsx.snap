// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="formInputs row2"
  >
    <div
      class="leftPart"
    >
      <div
        class="inputWrapper"
      >
        <input
          class="input"
          placeholder="請輸入"
          type="text"
          value=""
        />
      </div>
      <div
        class="inputWrapper"
      >
        <input
          class="input"
          placeholder="請輸入"
          type="number"
          value="0"
        />
      </div>
      <div
        class="inputWrapper"
      >
        <input
          class="input"
          disabled=""
          placeholder="請輸入"
          type="text"
          value=""
        />
      </div>
      <div
        class="passwordInput"
      >
        <div
          class="inputWrapper"
        >
          <input
            class="input"
            placeholder="密碼"
            type="password"
            value=""
          />
        </div>
        <div
          class="icon"
          role="button"
          tabindex="0"
        >
          <test-file-stub />
        </div>
      </div>
      <div
        class="row2"
      >
        <div
          class="inputWrapper"
        >
          <input
            class="input"
            placeholder="請輸入"
            type="text"
            value=""
          />
        </div>
        <div
          class="inputWrapper"
        >
          <input
            class="input"
            placeholder="請輸入"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="textareaWrapper textarea"
      >
        <textarea
          class="textarea"
          cols="10"
          placeholder="請輸入"
          rows="10"
        />
      </div>
    </div>
    <div
      class="rightPart"
    >
      <div
        class="dropdownWrapper"
      >
        <div
          role="button"
          tabindex="0"
        >
          <div
            class="dropdownSelectField"
          >
            <input
              disabled=""
              type="text"
              value="請選擇"
            />
            <div
              class="icon"
            >
              <test-file-stub />
            </div>
          </div>
        </div>
      </div>
      <div
        class="radioGroup radioGroup"
      >
        <div
          class="radioWrapper radio"
        >
          <input
            checked=""
            id="M"
            name="radioInput"
            type="radio"
          />
          <label
            for="M"
          >
            男
          </label>
        </div>
        <div
          class="radioWrapper radio"
        >
          <input
            id="F"
            name="radioInput"
            type="radio"
          />
          <label
            for="F"
          >
            女
          </label>
        </div>
      </div>
      <div
        class="checkboxGroup checkboxGroup"
      >
        <div
          class="checkboxWrapper checkbox"
        >
          <input
            id="選項A"
            name="選項A"
            type="checkbox"
          />
          <label
            for="選項A"
          >
            選項A
          </label>
        </div>
        <div
          class="checkboxWrapper checkbox"
        >
          <input
            id="選項B"
            name="選項B"
            type="checkbox"
          />
          <label
            for="選項B"
          >
            選項B
          </label>
        </div>
        <div
          class="checkboxWrapper checkbox"
        >
          <input
            checked=""
            id="選項C"
            name="選項C"
            type="checkbox"
          />
          <label
            for="選項C"
          >
            選項C
          </label>
        </div>
      </div>
      <div
        class="rangeSlider"
      >
        <input
          max="100"
          min="0"
          step="0"
          type="range"
          value="0"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
